### ISSUE TITLE
Compiling for MacOS

### ISSUE THREAD

**Issue by mfdorst** [CSUS-LLVM/OptSched#5]
In compiling for MacOS, I found two compile errors.
One was with `O_LARGEFILE` - which as far as I can tell is a flag for opening files, which does not exist in the MacOS libraries.
I found [this StackOverflow answer](https://stackoverflow.com/a/3425945/1291990) which seems to claim that this flag is used in the kernel, and isn't intended for use in applications.

The other issue was with the use of `constexpr`, in this declaration
```cpp
#define LSHPair std::pair<const char *, LISTSCHED_HEURISTIC>
static constexpr LSHPair HeuristicNames[] = {
    LSHPair("CP", LSH_CP),    LSHPair("LUC", LSH_LUC),
    LSHPair("UC", LSH_UC),    LSHPair("NID", LSH_NID),
    LSHPair("CPR", LSH_CPR),  LSHPair("ISO", LSH_ISO),
    LSHPair("SC", LSH_SC),    LSHPair("LS", LSH_LS),
    LSHPair("LLVM", LSH_LLVM)};
```
The error I got was
```
Constexpr variable 'HeuristicNames' must be initialized by a constant expression
```

I "fixed" these errors by simply removing the offending `O_LARGEFILE` and `constexpr`, but I have no idea what ramifications this may have.

**Comment by mfdorst** [comment]
mfdorst/OptSched@b6f67ee

**Comment by mfdorst** [comment]
I found that I was compiling with C++11, and once I moved to C++14, the `constexpr` error went away. This is because `std::pair` is not `constexpr` in C++11, but is in C++14.
